app/scripts/directives/multiSelectMenu.js

```diff
@@ -58,17 +58,17 @@ angular.module('gsUiInfraApp')
                 }
 
                 /**
-                 * Bind data to ng-model
+                 * call onchange function when model change
                  */
-                $scope.$watch('selected', function (newValue) {
-                    if (newValue) {
-
-                        setValue(newValue);
-                        if(angular.isFunction($scope.onchange)) {
-                            $scope.onchange({filter: newValue});
+                if(angular.isFunction($scope.onchange)) {
```

Comment:
The onchange attribute was already exist but it did not work. If we would use ng-change than there is no need for 'onchange' attribute at all, So Instead of fixing i'll just remove it and reopen PR